assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * core/widgets data store: widgets info.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { WIDGET_STYLES } from './constants';
+
+/**
+ * Store our widget components by registry, then by widget `slug`. We do this because
+ * we can't store React components in our data store.
+ *
+ * @private
+ * @since n.e.x.t
+ */
+export const WidgetComponents = {};
+
+const ASSIGN_WIDGET_AREA = 'ASSIGN_WIDGET_AREA';
+const REGISTER_WIDGET_AREA = 'REGISTER_WIDGET_AREA';
+
+export const INITIAL_STATE = {
+	areas: {},
+	contexts: {},
```

Comment:
Actually, can we name this `contextAssignments` or `contextAreaAssignments` or so? This is a more clear name - the current name looks like this would be for registered contexts, which it is not.

This becomes a more apparent problem for widgets and widget areas where we'll need to differentiate between `widgets` and `areaAssignments` / `areaWidgetAssignments` (see below).